### DISCUSSION TITLE
How to release dashboards to a production environment?

### DISCUSSION THREAD

**Discussion by ghost** [apache/superset#25789] [Q&A / Help]
Can you tell me how to release roles, dashboards, charts, datasets, etc. created in Apache Superset in a development environment to a production environment?

I exported the dashboard created in the development environment and imported it into the production environment, but Cross Filter Scoping was applied to all panels without permission, and the filter settings applied to individual panels were not imported.  
Is there a way to create the same dashboard in the production environment as in the development environment?  
Or is there an error in the concept of staging from a development environment to a production environment, as in conventional AP development?  

Here are the situations of the development and production environments:

- Apache Superset v2.1.0 is built using ECS Fargate on both development and production environments.
- Workers create roles, dashboards, charts, datasets, etc. in the development environment and test them.
- In the production environment, we want to display roles, dashboards, charts, datasets, etc. created in the development environment for customers to view.

Here are some things I want to achieve:

- The Athena database connection destination is different for each development and production environment. When releasing dashboards, etc., I want to change the connection destination and schema settings to match those of the production environment.
- Development is ongoing in the development environment, and it includes dashboards that should not be viewable in the production environment.
- We plan to release dashboards from the development environment to the production environment every few months.
- Users registered in the development and production environments are different. Ideally, I would like to keep logs and histories associated with users without deleting them by release.

Here are some things I tried:

- The following problems occurred with dashboard export/import, which did not meet my expectations:
  - The filter settings for individual panels of the dashboard were not imported.
  - When trying to update an already released dashboard, the dataset was not updated with new data and remained old.
- I exported a dump of the database where Apache Superset's dashboard information is stored and imported it by replacing the data source and schema settings. However, the following problems occurred and my expectations were not met:
  - Dashboards under development were displayed.
  - Users who should exist in the production environment along with their logs/histories disappeared.

Thanks.

**Comment by toransahu** [comment]
+1
